src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -421,6 +421,132 @@ void remove_redundant_reorders::run(program& p) {
             }
         }
     }
+
+    // Fuse reorder which only changes data precision
+    itr = p.get_processing_order().begin();
+    if (enable_reorder_fusing) {
+        while (itr != p.get_processing_order().end()) {
+            auto node_ptr = *itr++;
+            if (!node_ptr->is_type<reorder>())  // only care for reorders
+                continue;
+
+            auto& node = node_ptr->as<reorder>();
+            if (node.get_users().size() != 1 || node.is_output() || node.has_fused_primitives() ||
+                node.has_mean() || !node.get_primitive()->subtract_per_feature.empty()) {
+                continue;
+            }
+
+            auto& prev = node.input();
+            auto next_ptr = node.get_users().front();
+
+            auto is_simple = [](format fmt) {
+                return fmt == format::bfyx || fmt == format::byxf || fmt == format::yxfb;
+            };
+
+            auto prev_layout = prev.get_output_layout();
+            auto output_layout = node.get_output_layout();
+            auto next_layout = next_ptr->get_output_layout();
+
+            auto prev_simple = is_simple(prev_layout.format);
+            auto next_simple = is_simple(next_layout.format);
+            auto fmt_prev = prev_layout.format;
+            auto fmt_next = next_layout.format;
+            auto prev_dt = prev_layout.data_type;
+            auto next_dt = next_layout.data_type;
+
+            // Fuse reorder into next node
+            if (next_ptr->is_type<eltwise>() || next_ptr->is_type<permute>() || next_ptr->is_type<fully_connected>()) {
+                if (prev_dt == output_layout.data_type || fmt_prev != output_layout.format)
+                    continue;
+
+                bool can_fuse_reorder = false;
+                if (next_ptr->is_type<eltwise>() && (prev_simple && next_simple)) {
+                    // If accumulate type of eltwise is impacted from low precision to high precision, do not fuse to avoid accuracy degradation.
+                    if ((data_type_traits::size_of(prev_dt) >= data_type_traits::size_of(next_dt))
```

Comment:
I think `data_type_traits::size_of(next_dt)` should be `data_type_traits::size_of(output_layout.data_type)` because we are checking input precision of eltwise